Public/Src/Engine/Processes/SandboxedProcess.cs

```diff
@@ -69,6 +69,8 @@ public sealed class SandboxedProcess : ISandboxedProcess
         private TaskSourceSlim<bool> m_standardInputTcs;
         private readonly PathTable m_pathTable;
         private readonly string[] m_allowedSurvivingChildProcessNames;
+        private ulong m_maxPeakWorkingSet;

```

Comment:
Did you consider using `PerformanceCollector.Aggregation` here instead of `ulong`?  In that case you could make this field `readonly` and you wouldn't have to manually compute the maximum every time (the `Aggregation` object would take care of that)